xapian-core/HACKING

```diff
@@ -64,7 +64,7 @@ On Mac OS X, if you're using macports you'll want the following:
 
 If you're using homebrew you'll want the following::
 
-    brew install libmagic pcre
+    brew install libmagic pcre pkgconfig
```

Comment:
Updating the documentation of what packages to install is useful, but...

* This change doesn't belong in a commit described as "bootstrap: Add pkg.m4 directories on Mac OS".  Each commit should do one thing, and while this is somewhat connected, it's really a separate change - "Add pkg-config to required packages for build" or something like that.

* It's also not the complete change for that - the `pkg-config` should be added to the list of packages needed for Omega on Ubuntu and Debian.

* And `HACKING` is a legacy document we're in the process of getting rid of - the preferred location for listing packages to install is now: https://github.com/xapian/xapian-developer-guide/blob/master/getting-started/index.rst

It's OK to update HACKING, but equivalent changes should be made there too.  And if only one gets updated, it's better for it to be the developer guide.

So I'm going to revert the change to HACKING and get the other part merged.